torch/utils/_benchmark/utils/timer.py

```diff
@@ -75,7 +76,27 @@ def repeat(self, repeat=-1, number=-1):
     def autorange(self, callback=None):
         raise NotImplementedError("See `Timer.blocked_autorange.`")
 
-    def blocked_autorange(self, callback=None, min_run_time=0.2):
+    def adaptive_autorange(self, threshold=0.1, max_run_time=10.0, callback=None,
+            min_run_time=0.1):
+        start_time = time.time()
+        number = self._estimate_block_size(min_run_time=0.05)
+        measure = None
+        times = []
+        rounds = 0
+        with common.set_torch_threads(self._num_threads):
+            while time.time() - start_time < max_run_time:
```

Comment:
This will also include time spent in `callback`, `_construct_measurement`, etc.

Reply:
Oh, I see I misunderstood. I read this min_time as "Get back to be with answer in this time" as opposed to "collect this much signal"